src/network/taraxa_capability.cpp

```diff
@@ -400,21 +425,36 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
       break;
     }
     case GetBlocksPacket: {
-      LOG(log_dg_dag_sync_) << "Received GetBlocksPacket with " << _r.itemCount() << " known blocks";
-      std::unordered_set<blk_hash_t> known_non_finalized_blocks;
-      for (auto hash : _r) {
-        known_non_finalized_blocks.insert(blk_hash_t(hash));
-      }
+      std::unordered_set<blk_hash_t> blocks_hashes;
       std::vector<std::shared_ptr<DagBlock>> dag_blocks;
+      auto mode = static_cast<GetBlocksPacketRequestType>(_r[0].toInt<unsigned>());
+
+      if (mode == MissingHashes)
+        LOG(log_dg_dag_sync_) << "Received GetBlocksPacket with " << _r.itemCount() - 1 << " missing blocks";
+      else
+        LOG(log_dg_dag_sync_) << "Received GetBlocksPacket with " << _r.itemCount() - 1 << " known blocks";
+
+      for (size_t i = 1; i < _r.itemCount(); ++i) {
+        blocks_hashes.emplace(_r[i]);
```

Comment:
lets use iteration instead of random access as rlp does not support random access

Reply:
so why do not we remove operator[] from RLP?

Reply:
well my bad it supports it, but it is highly inefficient as it iterates the whole rlp from beginning when using [] until it gets to the index